cmd/lncli/cmd_invoice.go

```diff
@@ -2,13 +2,73 @@ package main
 
 import (
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"strconv"
 
 	"github.com/lightningnetwork/lnd/lnrpc"
 	"github.com/urfave/cli"
 )
 
+// newHopHint creates a new lnrpc.HopHint from the provided parameters.
+func newHopHint(nodeID string, chanID uint64, feeBaseMsat uint32,
+	feeProportionalMillionths uint32, cltvExpiryDelta uint32) (
+	*lnrpc.HopHint, error,
+) {
+
+	if nodeID == "" {
+		return nil, fmt.Errorf("node_id is missing in route hint")
+	}
+
+	if chanID == 0 {
+		return nil, fmt.Errorf("chan_id is missing or invalid in " +
+			"route hint")
+	}
```

Comment:
do we actually need to check these here? wont LND just fail later on?

Reply:
Good point. Will rely on the rpc for the validation 👍